platforms/BUILD

```diff
@@ -1,11 +1,49 @@
-# platforms/BUILD
+# Define all of the platform constraints here because bazel will only
+# automatically resolve a toolchain if all contraints for the target platform
+# match, including our contraints for jetpack and nvidia
 
 package(default_visibility = ["//visibility:public"])
 
+constraint_setting(name = "jetpack_version")
+
+constraint_setting(name = "nvidia_platform")
+
+constraint_value(
+    name = "jetpack_512",
+    constraint_setting = ":jetpack_version",
+    visibility = ["//visibility:public"],
+)
+
+constraint_value(
+    name = "jetpack_62",
+    constraint_setting = ":jetpack_version",
+    visibility = ["//visibility:public"],
+)
+
+constraint_value(
+    name = "tegra",
+    constraint_setting = ":nvidia_platform",
+    visibility = ["//visibility:public"],
+)
+
+platform(
+    name = "tegra_jetpack_512",
+    constraint_values = [
+        "@platforms//os:linux",
```

Comment:
Conceptually, the toolchain would only need to have the os/linux and cpu/arm64 constraint right? I would have hoped the actual platform definitions like `tegra_jetpack_512` wouldn't have to live here since it's defining constraints that are more narrow than what the toolchain provides.